src/loaders/bitmapFontParser.js

```diff
@@ -51,68 +107,23 @@ module.exports = function ()
         if (xmlUrl && xmlUrl.charAt(xmlUrl.length - 1) !== '/') {
             xmlUrl += '/';
         }
-
-        var textureUrl = xmlUrl + resource.data.getElementsByTagName('page')[0].getAttribute('file');
-        var loadOptions = {
-            crossOrigin: resource.crossOrigin,
-            loadType: Resource.LOAD_TYPE.IMAGE
-        };
-
-        // load the texture for the font
-        this.add(resource.name + '_image', textureUrl, loadOptions, function (res)
-        {
-            var data = {};
-            var info = resource.data.getElementsByTagName('info')[0];
-            var common = resource.data.getElementsByTagName('common')[0];
-
-            data.font = info.getAttribute('face');
-            data.size = parseInt(info.getAttribute('size'), 10);
-            data.lineHeight = parseInt(common.getAttribute('lineHeight'), 10);
-            data.chars = {};
-
-            //parse letters
-            var letters = resource.data.getElementsByTagName('char');
-
-            for (var i = 0; i < letters.length; i++)
-            {
-                var charCode = parseInt(letters[i].getAttribute('id'), 10);
-
-                var textureRect = new core.math.Rectangle(
-                    parseInt(letters[i].getAttribute('x'), 10),
-                    parseInt(letters[i].getAttribute('y'), 10),
-                    parseInt(letters[i].getAttribute('width'), 10),
-                    parseInt(letters[i].getAttribute('height'), 10)
-                );
-
-                data.chars[charCode] = {
-                    xOffset: parseInt(letters[i].getAttribute('xoffset'), 10),
-                    yOffset: parseInt(letters[i].getAttribute('yoffset'), 10),
-                    xAdvance: parseInt(letters[i].getAttribute('xadvance'), 10),
-                    kerning: {},
-                    texture: core.utils.TextureCache[charCode] = new core.Texture(res.texture.baseTexture, textureRect)
-
-                };
-            }
-
-            //parse kernings
-            var kernings = resource.data.getElementsByTagName('kerning');
-            for (i = 0; i < kernings.length; i++)
-            {
-                var first = parseInt(kernings[i].getAttribute('first'), 10);
-                var second = parseInt(kernings[i].getAttribute('second'), 10);
-                var amount = parseInt(kernings[i].getAttribute('amount'), 10);
-
-                data.chars[second].kerning[first] = amount;
-
-            }
-
-            resource.bitmapFont = data;
-
-            // I'm leaving this as a temporary fix so we can test the bitmap fonts in v3
-            // but it's very likely to change
-            extras.BitmapText.fonts[data.font] = data;
-
+        var file = resource.data.getElementsByTagName('page')[0].getAttribute('file');
+        if (utils.TextureCache[file]) {
```

Comment:
Yes it is relative to the XML, meaning this check will be checking for `font.png` which will never have been loaded before, or worse it will have been loaded before but it wont be what you expect.

Consider:

```
/
`--font.png
    |--bitmapFonts/
    |   |--font.xml
    |   `--font.png
```

If you load `/font.png` then `/bitmapFonts/font.xml` the XML load will check for `font.png` and actually get back `/font.png` when you wanted `/bitmapFonts/font.png`.

The reason the code you have here doesn't work is *because* the url is relative to itself. The cache will be keyed by url of the item that was loaded (see [textureParser.js][tp]). Does that make sense?

[tp]: https://github.com/GoodBoyDigital/pixi.js/blob/dev/src/loaders/textureParser.js#L12